client/my-sites/stats/activity-log/index.jsx

```diff
@@ -234,17 +234,17 @@ class ActivityLog extends Component {
 		const logsGroupedByDay = map(
 			groupBy(
 				logsForMonth,
-				log => applySiteOffset( moment.utc( log.ts_utc ) ).format( 'LL' )
+				log => applySiteOffset( moment.utc( log.ts_utc ) ).startOf( 'day' ).valueOf()
 			),
-			( daily_logs, day ) => (
+			( daily_logs, tsDayStart ) => (
 				<ActivityLogDay
 					allowRestore={ !! isPressable }
 					isRewindActive={ isRewindActive }
-					key={ day }
+					key={ tsDayStart }
 					logs={ daily_logs }
 					requestRestore={ this.handleRequestRestore }
 					siteId={ siteId }
-					day={ day }
+					tsDayStart={ +tsDayStart }
```

Comment:
I want to make sure we're getting this right. The start of a day _should be_ localized to site time, as that _does_ change:

```js
moment.utc( ts ).startOf('day').valueOf() // 0
moment.utc( ts ).tz('Asia/Bangkok').startOf('day').valueOf() // -25200000
moment.utc( ts ).utcOffset(12).startOf('day').valueOf() // -43200000
```

I what we're doing is correct, but please double check me 🙂